src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +19,45 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert phone to DB " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> findAllPhonesQuery = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> phoneRoot = findAllPhonesQuery.from(Phone.class);
+            List<Predicate> criterias = new ArrayList<>();
+            for (Map.Entry<String, String[]> currentPair : params.entrySet()) {
+                CriteriaBuilder.In<String> currentCriteria =
+                        criteriaBuilder.in(phoneRoot.get(currentPair.getKey()));
+                for (String currentValue : currentPair.getValue()) {
+                    currentCriteria.value(currentValue);
+                }
+                criterias.add(currentCriteria);
+            }
+            findAllPhonesQuery.where(criteriaBuilder.and(criterias.toArray(new Predicate[0])));
```

Comment:
you can use method reference, so that there are no magic numbers
```suggestion
            findAllPhonesQuery.where(criteriaBuilder.and(criterias.toArray(Predicate[]::new)));
```